qcelemental/models/results.py

```diff
@@ -97,6 +98,107 @@ def __str__(self):
+                                                   description="Beta-spin effective core (one-electron) Hamiltonian ")
+
+    # SCF Results
+    scf_orbitals_a: Optional[Array[float]] = Schema(None, description="SCF alpha-spin orbitals.")
+    scf_orbitals_b: Optional[Array[float]] = Schema(None, description="SCF beta-spin orbitals.")
+    scf_density_a: Optional[Array[float]] = Schema(None, description="SCF alpha-spin density matrix.")
+    scf_density_b: Optional[Array[float]] = Schema(None, description="SCF beta-spin density matrix.")
+    scf_fock_a: Optional[Array[float]] = Schema(None, description="SCF alpha-spin Fock matrix.")
+    scf_fock_b: Optional[Array[float]] = Schema(None, description="SCF beta-spin Fock matrix.")
+    scf_eigenvalues_a: Optional[Array[float]] = Schema(None, description="SCF alpha-spin eigenvalues.")
+    scf_eigenvalues_b: Optional[Array[float]] = Schema(None, description="SCF beta-spin eigenvalues.")
+    scf_occupations_a: Optional[Array[float]] = Schema(None, description="SCF alpha-spin occupations.")
+    scf_occupations_b: Optional[Array[float]] = Schema(None, description="SCF beta-spin occupations.")
+
+    # Return results, must be defined last
+    orbitals_a: Optional[str] = Schema(None, description="Index to the alpha-spin orbitals of the primary return.")
+    orbitals_b: Optional[str] = Schema(None, description="Index to the beta-spin orbitals of the primary return.")
+    density_a: Optional[str] = Schema(None, description="Index to the alpha-spin density of the primary return.")
+    density_b: Optional[str] = Schema(None, description="Index to the beta-spin density of the primary return.")
+    fock_a: Optional[str] = Schema(None, description="Index to the alpha-spin Fock matrix of the primary return.")
+    fock_b: Optional[str] = Schema(None, description="Index to the beta-spin Fock matrix of the primary return.")
+    eigenvalues_a: Optional[str] = Schema(None,
+                                          description="Index to the alpha-spin eigenvalues of the primary return.")
+    eigenvalues_b: Optional[str] = Schema(None,
+                                          description="Index to the beta-spin eigenvalues of the primary return.")
+    occupations_a: Optional[str] = Schema(
+        None, description="Index to the alpha-spin orbital eigenvalues of the primary return.")
+    occupations_b: Optional[str] = Schema(
+        None, description="Index to the beta-spin orbital eigenvalues of the primary return.")
+
+    class Config(ProtoModel.Config):
+        force_skip_defaults = True
+
+    @validator('scf_eigenvalues_a', 'scf_eigenvalues_b', 'scf_occupations_a', 'scf_occupations_b', whole=True)
+    def _assert1d(cls, v, values):
+        bas = values.get("basis", None)
+
+        # Do not raise multiple errors
+        if bas is None:
+            return v
+
+        try:
+            v = v.reshape(bas.nbf)
+        except (ValueError, AttributeError):
```

Comment:
Two things: 

1. ValueError message should be "shape (nbf)'. 
2. is `nbf` guaranteed to be `nmo` for SCF, or are there edge cases?

Reply:
Could point, I made a new validator for `(nbf, -1)` for now

Reply:
This gets tricky as then `nbf` may change *again* after SCF, after MP2, and even after CCSD...

Reply:
I understand how nmo would change, but why would nbf change?

Reply:
Sorry, replace above `nbf` with `nmo`.

Reply:
I think this okay; we decided that nmo was allowed to be anything and that we wouldn't validate it.